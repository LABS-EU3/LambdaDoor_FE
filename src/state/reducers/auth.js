import * as types from '../types';

const initialState = {
  isLoggingIn: false,
  isLoggedIn: false,
};

export const authState = (state = initialState, action) => {
  switch (action.type) {
    case types.LOG_IN_USER:
      return {
        ...state,
        isLoggingIn: true,
      };
    case types.GET_COMPANY_REVIEWS_SUCCESS:
      return {
        ...state,
        isLoggedIn: true,
      };
    case types.GET_COMPANY_REVIEWS_FAILURE:
      return state;
    default:
      return state;
  }
};
