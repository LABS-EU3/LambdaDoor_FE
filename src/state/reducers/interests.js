import * as types from '../types';
import { allInterests } from '../../utils/data';

const initialState = {
  interests: [],
};

// eslint-disable-next-line import/prefer-default-export
export const interestReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_INTERESTS_SUCCESS:
      return {
        ...state,
        isLoading: false,
        interests: action.payload,
      };

    case types.DELETE_INTEREST:
      console.log(action.payload);
      return {
        ...state,
        interests: state.interests.filter(
          interest => interest.id !== action.payload
        ),
      };

    case types.ADD_INTEREST:
      return {
        ...state,
        interests: [
          ...state.interests,
          { id: action.payload, interest: allInterests[action.payload] },
        ],
      };
    default:
      return state;
  }
};
