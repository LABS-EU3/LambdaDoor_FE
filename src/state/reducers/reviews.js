import * as types from '../types';

const initialState = {
  isFetching: false,
  reviews: [],
};

export const companyReviews = (state = {}, action) => {
  switch (action.type) {
    case types.GET_COMPANY_REVIEWS:
      return {};
    case types.GET_COMPANY_REVIEWS_SUCCESS:
      return {
        ...state,
        message: action.payload.message,
      };
    case types.GET_COMPANY_REVIEWS_FAILURE:
      return {};
    default:
      return state;
  }
};

export const salaryReviews = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_SALARY_REVIEWS:
      return { ...state, isFetching: true };
    case types.GET_SALARY_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: [...state.reviews, action.payload],
        isFetching: false,
      };
    case types.GET_SALARY_REVIEWS_FAILURE:
      return { ...state, isFetching: false };
    default:
      return state;
  }
};

export const interviewReviews = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_INTERVIEW_REVIEWS:
      return { ...state, isFetching: true };
    case types.GET_INTERVIEW_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: [...state.reviews, action.payload],
      };
    case types.GET_INTERVIEW_REVIEWS_FAILURE:
      return { ...state, isFetching: false };
    default:
      return state;
  }
};

export const reviewsError = (state = null, action) => {
  switch (action.type) {
    case types.GET_COMPANY_REVIEWS_FAILURE:
    case types.GET_SALARY_REVIEWS_FAILURE:
    case types.GET_INTERVIEW_REVIEWS_FAILURE:
      return action.payload;
    case types.GET_COMPANY_REVIEWS:
    case types.GET_SALARY_REVIEWS:
    case types.GET_INTERVIEW_REVIEWS:
      return null;
    case types.GET_COMPANY_REVIEWS_SUCCESS:
    case types.GET_SALARY_REVIEWS_SUCCESS:
    case types.GET_INTERVIEW_REVIEWS_SUCCESS:
      return null;
    default:
      return state;
  }
};
