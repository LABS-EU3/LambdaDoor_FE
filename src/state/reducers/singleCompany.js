import * as types from '../types';

const initialState = {
  isFetching: false,
  reviews: {
    companyReview: [],
    // salaryReview: [],
    // interviewProcess: [],
  },
};

const companyReviewsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_SINGLE_COMPANY_REVIEWS:
      return { ...state, isFetching: true };

    case types.GET_SINGLE_COMPANY_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: {
          ...state.reviews,
          companyReview: action.payload,
        },
      };

    case types.GET_COMPANY_REVIEWS_FAILURE:
      return { ...state, isFetching: false };

    default:
      return state;
  }
};

export default companyReviewsReducer;
