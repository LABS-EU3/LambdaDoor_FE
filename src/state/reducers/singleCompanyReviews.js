/* eslint-disable import/prefer-default-export */
import * as types from '../types';

const initialState = {
  isFetching: false,
  reviews: {
    company: [],
    salary: [],
    interview: [],
  },
};

export const singleCompanyReviewsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_SINGLE_COMPANY_INTERVIEW_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: {
          ...state.reviews,
          interview: action.payload,
        },
      };

    default:
      return state;
  }
};
