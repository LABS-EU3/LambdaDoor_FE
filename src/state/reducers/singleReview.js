import * as types from '../types';

const initialState = {
  isFetching: false,
  reviews: {
    companyReview: [],
  },
};

const singleReviewsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_SINGLE_REVIEWS:
      return { ...state, isFetching: true };

    case types.GET_SINGLE_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: {
          ...state.reviews,
          companyReview: action.payload,
        },
      };

    case types.GET_SINGLE_REVIEWS_FAILURE:
      return { ...state, isFetching: false };

    default:
      return state;
  }
};

export default singleReviewsReducer;
