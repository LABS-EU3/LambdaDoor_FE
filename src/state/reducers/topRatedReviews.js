import * as types from '../types';

const initialState = {
  isfetching: false,
  topRatedReviews: [],
};
// eslint-disable-next-line import/prefer-default-export
export const topRatedReviewsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_TOP_RATED:
      return { ...state, isFetching: true };
    case types.GET_TOP_RATED_SUCCESS:
      return {
        isfetching: false,
        topRatedReviews: action.payload,
      };
    case types.GET_TOP_RATED_FAILURE:
      return { ...state, isFetching: false };
    default:
      return state;
  }
};
