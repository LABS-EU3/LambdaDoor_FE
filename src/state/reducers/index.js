import { combineReducers } from 'redux';
import { reviewsReducer, reviewsError } from './reviews';
import { authState } from './auth';
import { interestReducer } from './interests';
import { companiesReducer } from './companies';
import { jobrolesReducer } from './jobroles';
import companyReviewsReducer from './singleCompany';

const rootReducer = combineReducers({
  reviews: reviewsReducer,
  reviewsError: reviewsError,
  authState: authState,
  interests: interestReducer,
  companies: companiesReducer,
  jobroles: jobrolesReducer,
  singleCompanyReviews: companyReviewsReducer,
});

export default rootReducer;
