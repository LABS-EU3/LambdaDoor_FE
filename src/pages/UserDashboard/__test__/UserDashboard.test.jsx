import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import reducers from '../../../state/reducers/index';
import { renderWithRedux, TestingRouter } from '../../../utils/testingHelpers';
import ConnectedUserDashboard, { UserDashboard } from '../UserDashboard';


const state = reducers(undefined, {});

describe('UserDashboard', () => {
  it('should equal true if true', () => {
    expect(true).toEqual(true);
  });
});
