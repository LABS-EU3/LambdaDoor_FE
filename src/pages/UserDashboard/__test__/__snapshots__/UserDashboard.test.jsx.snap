// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedUserDashboard renders correctly 1`] = `
<body>
  <div>
    <div>
      <h1
        class="sc-htoDjs awifR"
      >
        Latest Reviews
      </h1>
      <div
        class="sc-ifAKCX hGuNVX"
      >
        <div
          class="empty-state"
        >
          <div
            class="ant-spin ant-spin-spinning"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
